test/test_torch.py

```diff
@@ -2357,6 +2358,53 @@ def make_prob_dist(shape, is_contiguous):
     def test_multinomial(self):
         self._test_multinomial(self, torch.FloatTensor)
 
+    def _spawn_method(self, method, arg):
+        try:
+            mp.set_start_method('spawn')
+        except RuntimeError:
+            pass
+        with mp.Pool(1) as pool:
+            self.assertTrue(pool.map(method, [arg]))
+
+    @staticmethod
+    def _test_multinomial_invalid_probs(probs):
+        try:
+            torch.multinomial(probs.to('cpu'), 1)
+            return False  # Should not be reached
+        except RuntimeError as e:
+            return 'invalid multinomial distribution' in str(e)
+
+    @unittest.skipIf(sys.version_info[0] == 2,
+                     "spawn start method is not supported in Python 2, \
+                     but we need it for for testing failure case for CPU RNG on Windows")
+    def test_multinomial_invalid_probs(self):
```

Comment:
If possible we should try not to use multiprocessing for testing. For tests on the CPU, you can just use `self.assertRaisesRegex`.

Reply:
On Windows it will give `RuntimeError: device or resource busy: device or resource busy` error when we try to use the CPU RNG again after one failure case. I haven't figured out the reason for it, but for now we probably need to treat it the same way we do the CUDA tests.

Reply:
That is bizarre... does the same thing happen if you put in negative values for the distribution on windows?

Reply:
Yeah it will look like this:
```
>>> import torch
>>> torch.multinomial(torch.Tensor([0, -1]).to('cpu'), 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: invalid argument 2: invalid multinomial distribution (encountering
 probability entry < 0) at d:\willfeng\pytorch\aten\src\th\generic/THTensorRando
m.cpp:317
>>> torch.multinomial(torch.Tensor([0, -1]).to('cpu'), 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: device or resource busy: device or resource busy
```